pygem/setup/config.py

```
@@ -7,42 +7,93 @@
     
-    if (not isfile) or (overwrite):
-        os.makedirs(basedir, exist_ok=True)  # Ensure the base directory exists
-        shutil.copy(source_config_file, config_file)  # Copy the file
-        print(f"Copied default configuration to {config_file}")
+    def ensure_config(self, overwrite=False):
+        """Ensure the configuration file exists"""
+        if os.path.isfile(self.config_path) and not overwrite:
+            return
+        
+        if os.path.isfile(self.config_path) and overwrite:
+            overwrite = self._prompt_overwrite()
+        
+        if not os.path.isfile(self.config_path) or overwrite:
+            os.makedirs(self.base_dir, exist_ok=True)
+            shutil.copy(self.source_config_path, self.config_path)
+            print(f"Copied default configuration to {self.config_path}")
     
-def read_config():
-    """Read the configuration file and return its contents as a dictionary."""
-    with open(config_file, 'r') as f:
-        config = yaml.safe_load(f)  # Use safe_load to avoid arbitrary code execution
-    return config
\ No newline at end of file
+    def read_config(self):
+        """Read the configuration file and return its contents as a dictionary."""
+        with open(self.config_path, 'r') as f:
+            return yaml.safe_load(f)
+    
+    def update_config(self, updates):
+        """Update multiple keys in the YAML configuration file while preserving quotes and original types."""
+        yaml = YAML()
+        yaml.preserve_quotes = True  # Preserve quotes around string values
+
+        try:
+            with open(self.config_path, 'r') as file:
+                config = yaml.load(file)
+
+            for key, value in updates.items():
+                keys = key.split('.')
+                d = config
+                for k in keys[:-1]:
+                    if k not in d:
+                        print(f"Key '{key}' not found in the config. Skipping update.")
+                        break
```

I think this should raise an error since the user most likely made a mistake, so it doesn't make sense to continue the simulation, right?

---

Yes, that's a good point. I'll change to an exception. Thanks